robot/impl/local_robot.go

```diff
@@ -1053,7 +1053,7 @@ func (r *localRobot) Reconfigure(ctx context.Context, newConfig *config.Config)
 	// TODO(RSDK-1849): Make this non-blocking so other resources that do not require packages can run before package sync finishes.
 	// TODO(RSDK-2710) this should really use Reconfigure for the package and should allow itself to check
 	// if anything has changed.
-	err := r.packageManager.Sync(ctx, newConfig.Packages)
+	err := r.packageManager.Sync(ctx, newConfig.Packages, newConfig.Modules)
```

Comment:
Will this cause downloads on every call to `Reconfigure` even if the user's config hasn't changed at all?

Reply:
Nope -- that is the point of m.managedPackages and the check if the dir exists.

Package manager is a bit special in that it stores a fair bit of state inside itself.